src/main/java/dinkplugin/util/ItemSearcher.java

```diff
@@ -0,0 +1,142 @@
+package dinkplugin.util;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonParseException;
+import com.google.gson.reflect.TypeToken;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import okhttp3.Call;
+import okhttp3.Callback;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.io.IOException;
+import java.io.Reader;
+import java.lang.reflect.Type;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+@Slf4j
+@Singleton
+public class ItemSearcher {
+
+    private static final String RUNELITE_ITEM_CACHE = "https://static.runelite.net/cache/item/";
+    private final Map<String, Integer> itemIdByName = Collections.synchronizedMap(new HashMap<>());
+    private final CountDownLatch latch = new CountDownLatch(1);
+
+    @Inject
+    private OkHttpClient httpClient;
+
+    @Inject
+    private Gson gson;
+
+    @Nullable
+    @SneakyThrows
+    public Integer findItemId(String name) {
+        latch.await(5, TimeUnit.SECONDS);
+        return itemIdByName.get(name);
+    }
+
+    @Nullable
+    public String getItemImageUrl(String itemName) {
+        Integer itemId = findItemId(itemName);
+        return itemId != null ? ItemUtils.getItemImageUrl(itemId) : null;
+    }
+
+    @Inject
+    void init() {
+        queryNamesById()
+            .thenAcceptBothAsync(
+                queryNotedItemIds().exceptionally(throwable -> Collections.emptySet()),
+                this::merge
+            )
+            .exceptionally(e -> {
```

Comment:
@pajlada Would you prefer the `CompletableFuture`/`CountDownLatch` approach be replaced with `ScheduledExecutorService` (which uses our blocking impl under tests)?